iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessor.java

```diff
@@ -1893,23 +1762,96 @@ private void writeStoreToDisk(FileNodeProcessorStore fileNodeProcessorStore)
 
   private FileNodeProcessorStore readStoreFromDisk() throws FileNodeProcessorException {
 
-    synchronized (fileNodeRestoreFilePath) {
+    synchronized (fileNodeRestoreLock) {
       FileNodeProcessorStore processorStore;
       SerializeUtil<FileNodeProcessorStore> serializeUtil = new SerializeUtil<>();
       try {
         processorStore = serializeUtil.deserialize(fileNodeRestoreFilePath)
             .orElse(new FileNodeProcessorStore(false, new HashMap<>(),
                 new IntervalFileNode(OverflowChangeType.NO_CHANGE, null),
-                new ArrayList<IntervalFileNode>(), FileNodeProcessorStatus.NONE, 0));
+                new ArrayList<>(), FileNodeProcessorStatus.NONE, 0));
       } catch (IOException e) {
-        e.printStackTrace();
         throw new FileNodeProcessorException(e);
       }
       return processorStore;
     }
   }
 
-  public String getFileNodeRestoreFilePath() {
+  String getFileNodeRestoreFilePath() {
     return fileNodeRestoreFilePath;
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Do we really need to override the equals() and hashCode() method of this class? what the Sonar said? 
Remember that if we override the equals() and hashcode() functions like what you wrote, we have to notice ourselves we need to modify these two functions if we add (or remove) fields in this class..

Reply:
This is required by Sonar that any sub-classes with new fields should override equals().

Reply:
> This is required by Sonar that any sub-classes with new fields should override equals().

That is to say.... we have to override the equals() and hashCode() for ALL JAVA Files??

Reply:
Of course you may not need these fields to distinguish one object from another, but giving the equals() method explicitly shows that you know how to distinguish two objects.

Reply:
I mean sub-classes of classes DEFINED BY US. While I still think making all classes have its equals() is not a bad idea.